var/spack/repos/builtin/packages/py-psycopg2/package.py

```diff
@@ -12,14 +12,26 @@ class PyPsycopg2(PythonPackage):
     homepage = "https://psycopg.org/"
     pypi = "psycopg2/psycopg2-2.8.6.tar.gz"
 
+    version("2.9.6", sha256="f15158418fd826831b28585e2ab48ed8df2d0d98f502a2b4fe619e7d5ca29011")
     version("2.9.1", sha256="de5303a6f1d0a7a34b9d40e4d3bef684ccc44a49bbe3eb85e3c0bffb4a131b7c")
     version("2.8.6", sha256="fb23f6c71107c37fd667cb4ea363ddeb936b348bbd6449278eb92c189699f543")
     version("2.7.5", sha256="eccf962d41ca46e6326b97c8fe0a6687b58dfc1a5f6540ed071ff1474cea749e")
 
     # https://www.psycopg.org/docs/install.html#prerequisites
-    depends_on("python@3.6:3.9", type=("build", "link", "run"), when="@2.9:")
-    depends_on("python@2.7:2.8,3.4:3.8", type=("build", "link", "run"), when="@2.8")
-    depends_on("python@2.6:2.8,3.2:3.7", type=("build", "link", "run"), when="@:2.7")
+    # https://github.com/psycopg/psycopg2/blob/master/doc/src/install.rst
+    # https://www.psycopg.org/docs/news.html#news
+    # https://pypi.org/project/psycopg2/#history
+    depends_on("python@:3.11", when="@2.9.5:", type=("build", "link", "run"))
+    depends_on("python@:3.10", when="@2.9.1:2.9.4", type=("build", "link", "run"))
+    depends_on("python@:3.9", when="@2.8.6:2.9.0", type=("build", "link", "run"))
+    depends_on("python@:3.8", when="@2.8.4:2.8.5", type=("build", "link", "run"))
```

Comment:
```suggestion
    depends_on("python@:3.10", when="@2.9.2:2.9.4", type=("build", "link", "run"))
    depends_on("python@:3.9", when="@2.9:2.9.1", type=("build", "link", "run"))
    depends_on("python@:3.8", when="@2.8.4:2.8.6", type=("build", "link", "run"))
```
According to the install.rst:
- 2.9.1 has a python upper bound of 3.9 (see https://github.com/psycopg/psycopg2/blob/2_9_1/doc/src/install.rst)
- 2.8.6 has a python upper bound of 3.8 (see https://github.com/psycopg/psycopg2/blob/2_8_6/doc/src/install.rst)

Reply:
2.9.1 has [3.10 wheels](https://pypi.org/project/psycopg2/2.9.1/#files)

2.8.6 has [3.9 wheels](https://pypi.org/project/psycopg2/2.8.6/#files)

Reply:
I'm fine with changing this to match the docs if you want, but PyPI is where I got these specific versions from.

Reply:
No, it's fine. I did not see the wheels and did not understand that, with the comment to pypi, you meant that.